### PR TITLE
Implement GenericAssignment for DerivedTypes

### PR DESCRIPTION
**Pull request by czgdp1807**
https://github.com/lfortran/lfortran/commit/2cc6460b37b5545676178e4b7f7821356d774f8d
https://github.com/lfortran/lfortran/commit/aaea271d3bc3e826dec456bf5f3776a69d4aaec5
https://github.com/lfortran/lfortran/commit/6aa545c611beb0370b60a084542648d750e195ad

### CODE REVIEW

File: src/libasr/ASR.asdl [hunk 1]
```diff
@@ -107,8 +107,8 @@ symbol
         intent intent, expr? symbolic_value, expr? value, storage_type storage, ttype type,
         abi abi, access access, presence presence, bool value_attr)
     | ClassType(symbol_table symtab, identifier name, abi abi, access access)
-    | ClassProcedure(symbol_table parent_symtab, identifier name, identifier proc_name,
-        symbol proc, abi abi)
+    | ClassProcedure(symbol_table parent_symtab, identifier name, identifier? pass_argument,
```
Comment (certik) [inline src/libasr/ASR.asdl:110]:
```suggestion
    | ClassProcedure(symbol_table parent_symtab, identifier name, identifier? self_argument,
```

Comment (certik) [reply src/libasr/ASR.asdl:110]: also document every change / new arguments in `src/asr/asr_nodes/symbol_nodes/symbol.md`.

Comment (certik) [reply src/libasr/ASR.asdl:110]: Maybe you can use `expr?` here.

Comment (czgdp1807) [reply src/libasr/ASR.asdl]: Done.

**Review by certik — APPROVED** [review]

**Pull request merged**